src/Grids/grid_utils.jl

```diff
@@ -225,121 +226,98 @@ index_range_offset(::Colon, loc, topo, halo)          = - interior_parent_offset
 
-@inline cpu_face_constructor_x(grid) = Array(all_x_nodes(Face, grid)[1:grid.Nx+1])
-@inline cpu_face_constructor_y(grid) = Array(all_y_nodes(Face, grid)[1:grid.Ny+1])
-@inline cpu_face_constructor_z(grid) = Array(all_z_nodes(Face, grid)[1:grid.Nz+1])
+@inline cpu_face_constructor_x(grid) = Array(xnodes(grid, Face(); with_halos=true)[1:grid.Nx+1])
+@inline cpu_face_constructor_y(grid) = Array(ynodes(grid, Face(); with_halos=true)[1:grid.Ny+1])
+@inline cpu_face_constructor_z(grid) = Array(znodes(grid, Face(); with_halos=true)[1:grid.Nz+1])
 
-all_x_nodes(::Type{Nothing}, grid) = 1:1
-all_y_nodes(::Type{Nothing}, grid) = 1:1
-all_z_nodes(::Type{Nothing}, grid) = 1:1
+xnodes(grid, ::Nothing; kwargs...) = 1:1
+ynodes(grid, ::Nothing; kwargs...) = 1:1
+znodes(grid, ::Nothing; kwargs...) = 1:1
 
 """
-    xnodes(loc, grid, reshape=false)
-
-Return a view over the interior `loc=Center` or `loc=Face` nodes
-on `grid` in the ``x``-direction. For `Bounded` directions,
-`Face` nodes include the boundary points.
-
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `Nx×1×1`.
+    xnodes(grid, LX, LY, LZ, with_halos=false)
 
 See `znodes` for examples.
 """
-function xnodes(loc, grid; reshape=false)
-
-    x = view(all_x_nodes(loc, grid),
-             interior_indices(loc, topology(grid, 1), grid.Nx))
-
-    return reshape ? Base.reshape(x, length(x), 1, 1) : x
-end
+@inline xnodes(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = xnodes(grid, LX; kwargs...)
 
 """
-    ynodes(loc, grid, reshape=false)
-
-Return a view over the interior `loc=Center` or `loc=Face` nodes
-on `grid` in the ``y``-direction. For `Bounded` directions,
-`Face` nodes include the boundary points.
+    ynodes(grid, LX, LY, LZ, with_halos=false)
 
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `1×Ny×1`.
-
-See [`znodes`](@ref) for examples.
+See `znodes` for examples.
 """
-function ynodes(loc, grid; reshape=false)
-
-    y = view(all_y_nodes(loc, grid),
-             interior_indices(loc, topology(grid, 2), grid.Ny))
-
-    return reshape ? Base.reshape(y, 1, length(y), 1) : y
-end
+@inline ynodes(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = ynodes(grid, LY; kwargs...)
 
 """
-    znodes(loc, grid, reshape=false)
-
-Return a view over the interior `loc=Center` or `loc=Face` nodes
-on `grid` in the ``z``-direction. For `Bounded` directions,
-`Face` nodes include the boundary points.
-
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `1×1×Nz`.
-
-Examples
-========
+    znodes(grid, LX, LY, LZ, with_halos=false)
```

Comment:
This docstring seems incomplete. Needs explanation of what this function does, right?

Reply:
Added explanation